motr/setup.c

```diff
@@ -1473,342 +1473,26 @@ static int cs_be_dom_cfg_zone_pcnt_fill(struct m0_reqh          *reqh,
-
-		part_cfg->bpc_create_cfg = m0_alloc(strlen(rctx->rc_be_seg_path) +
-						    len);
-		M0_ASSERT(part_cfg->bpc_create_cfg != NULL);
-		sprintf(part_cfg->bpc_create_cfg, "%p %s %"PRIu64,
-			&rctx->rc_be.but_dom,
-			rctx->rc_be_seg_path,
-			sdev->sd_size);
-		part_cfg->bpc_init_cfg = part_cfg->bpc_create_cfg;
-		part_cfg->bpc_location =
-			(char*)cs_storage_partdom_location_gen(rctx->rc_be_seg_path,
-								&rctx->rc_be.but_dom);
-		part_cfg->bpc_dom_key = sdev->sd_dev_idx;
-
-
-		/* 1 chunk for partition table itself */
-		used_chunks = 1;
-
-		/** parition stob generic configuration */
-		part_cfg->bpc_part_mode_set = true;
-		proposed_chunk_size = rctx->rc_be_seg_size / def_dev_chunk_count;
-		part_cfg->bpc_chunk_size_in_bits =
-			align_chunk_size(proposed_chunk_size);
-		part_cfg->bpc_total_chunk_count =
-			rctx->rc_be_seg_size >> part_cfg->bpc_chunk_size_in_bits;
-
-		// seg0 configuration
-		part_cfg->bpc_stobs_cfg[M0_BE_DOM_PART_IDX_SEG0].bps_size_in_chunks = 1;
-		part_cfg->bpc_stobs_cfg[M0_BE_DOM_PART_IDX_SEG0].bps_id = M0_BE_PTABLE_ENTRY_SEG0;
-		part_cfg->bpc_stobs_cfg[M0_BE_DOM_PART_IDX_SEG0].bps_enble = true;
-		rctx->rc_be_seg0_path = rctx->rc_be_seg_path;
-		used_chunks += part_cfg->bpc_stobs_cfg[M0_BE_DOM_PART_IDX_SEG0].bps_size_in_chunks;
-		part_cfg->bpc_stobs_cfg[M0_BE_DOM_PART_IDX_SEG0].bps_create_cfg = 
-				cs_storage_partstob_location_gen(rctx->rc_be_seg0_path, "directio:false");
-		part_cfg->bpc_stobs_cfg[M0_BE_DOM_PART_IDX_SEG0].bps_init_cfg = 
-			part_cfg->bpc_stobs_cfg[M0_BE_DOM_PART_IDX_SEG0].bps_create_cfg;
-
-
-		/** seg1 configuration */
-		part_cfg->bpc_stobs_cfg[M0_BE_DOM_PART_IDX_SEG1].bps_enble = true;
-		part_cfg->bpc_stobs_cfg[M0_BE_DOM_PART_IDX_SEG1].bps_id = M0_BE_PTABLE_ENTRY_SEG1;
-		part_cfg->bpc_stobs_cfg[M0_BE_DOM_PART_IDX_SEG1].bps_size_in_chunks =
-			part_cfg->bpc_total_chunk_count - used_chunks;
-		rctx->rc_be_seg_size = m0_align(part_cfg->bpc_stobs_cfg[M0_BE_DOM_PART_IDX_SEG1].bps_size_in_chunks <<
-			part_cfg->bpc_chunk_size_in_bits, M0_BE_SEG_PAGE_SIZE);
-		part_cfg->bpc_stobs_cfg[M0_BE_DOM_PART_IDX_SEG1].bps_create_cfg = 
-				cs_storage_partstob_location_gen(rctx->rc_be_seg_path, "directio:false");
-		part_cfg->bpc_stobs_cfg[M0_BE_DOM_PART_IDX_SEG1].bps_init_cfg = 
-			part_cfg->bpc_stobs_cfg[M0_BE_DOM_PART_IDX_SEG1].bps_create_cfg;
-		// used_chunks += part_cfg->bpc_seg_size_in_chunks;
-
-	} else if (rctx->rc_be_seg0_path != NULL &&
-		   rctx->rc_be_seg_path != NULL &&
-		   rctx->rc_be_log_path != NULL) {
-
-		/**
-		 * Case 3: Single Data and Meta and log
-		 *         specified
-		 *
-		 *  partition stob is not needed
-		 */
-
+	    (!rctx->rc_stob.s_ad_disks_init))
 		return;
-	} else if (rctx->rc_be_seg_path == NULL &&
-		   dev_count > 1 ) {
 
-		/**
-		 * Case 5: Multiple Data and no Meta specified
-		 */
-		return;
-	}
+        /* if (cs_conf_part_config_get(rctx, part_cfg, false) != 0)
+		return ; */
 
+        if (cs_conf_part_config_get(rctx, part_cfg, true) != 0)
```

Comment:
return no useful here

Reply:
removed return and if condition.
commit 351218bfea6c47b27beb58f260b4485560c5872f